contributor-docs/style.md

```diff
@@ -0,0 +1,470 @@
+  - [Prefer using CSS Custom Properties to bridge between CSS and JavaScript instead of inline styles](#prefer-using-css-custom-properties-to-bridge-between-css-and-javascript-instead-of-inline-styles)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- prettier-ignore-end -->
+
+## React
+
+### Server-side rendering
+
+All components should be able to be rendered on the server. This means that the
+component must not access any JavaScript APIs that are only available in a
+browser.
+
+In addition, authors should consider how components will be hydrated as a result
+of being server-side rendered. In particular, components should not cause layout
+shifts as a result of being hydrated. Instead components should progressively
+enhance as JavaScript is loaded on the page.
+
+### Prefer managing focus through event handlers instead of effects
+
+When managing focus within a component, prefer focusing elements within event
+handlers as opposed to effects. This helps to ensure that focus is only being
+moved as a result of a user interaction. When focus is managed in an effect,
+there is a risk that focus will be moved due to an unrelated dependency in the
+effect dependency array being updated.
+
+### Prefer the `useControllableState` hook when authoring components that can be controlled or uncontrolled
+
+Components may be authored in a way that allows state values to be controlled or
+uncontrolled. For example:
+
+```tsx
+type ExampleProps = {
+  value?: string
+  defaultValue?: string
+  onChange({value}: {value: string}): void
+}
+
+function Example(props: ExampleProps) {
+  // ...
+}
+
+function UncontrolledExample() {
+  return (
+    <Example
+      defaultValue="Example default value"
+      onChange={({value}) => {
+        console.log(value)
+      }}
+    />
+  )
+}
+
+function ControlledExample() {
+  const [value, setValue] = React.useState('Example controlled value')
+  return (
+    <Example
+      value={value}
+      onChange={({value}) => {
+        setValue(value)
+      }}
+    />
+  )
+}
+```
+
+When building out components that can controlled or uncontrolled, prefer using
+the `useControllableState` hook. With the `Example` component above, this
+would look like:
+
+```tsx
+function Example({defaultValue, onChange, value: controlledValue}: ExampleProps) {
+  const [value, setValue] = useControllableState({
+    value: controlledValue,
+    defaultValue,
+    onChange: ({value}) => {
+      onChange({value})
+    },
+  })
+}
+```
+
+### Props
+
+#### Prefer naming opt-in `boolean` props so that applying them specifies the change that will occur
+
+<table>
+<thead><tr><th>Unpreferred</th><th>Preferred</th></tr></thead>
+<tbody>
+<tr><td>
+
+```tsx
+type Props = {}
+```
+
+</td><td>
+
+```tsx
+type Props = {
+  fullWidth?: boolean
+}
```

Comment:
I don't get this example 😅

Reply:
Super fair, let me think of a better one (or just remove it lol)